### PR TITLE
Document `Vector2.dot()` and `Vector3.dot()` more extensively

### PR DESCRIPTION
**Pull request by Calinou**
These methods are commonly used in games. It's time to make the documentation more explicit about them :)

### CODE REVIEW

File: doc/classes/Vector3.xml [hunk 1]
```diff
@@ -123,7 +123,10 @@
 			<argument index="0" name="b" type="Vector3">
 			</argument>
 			<description>
-				Returns the dot product with [code]b[/code].
+				Returns the dot product with vector [code]b[/code]. This can be used to compare the angle between two vectors. For example, this can be used to determine whether an enemy is facing the player.
+				The dot product will be [code]0[/code] for a straight angle (90 degrees), greater than 0 for angles narrower than 90 degrees and lower than 0 for angles wider than 90 degrees.
+				When using unit vectors, the result will always be between [code]-1[/code] (180 degree angle) and [code]1[/code] (0 degree angle).
+				[b]Note:[/b] [code]a.dot(b)[/code] is equivalent to [code]b.dot(a)[/code].
```
Comment (Calinou) [inline doc/classes/Vector3.xml:129]: Does this new description apply to Vector3 like it does to Vector2? I haven't thoroughly checked this.

Comment (aaronfranke) [reply doc/classes/Vector3.xml:129]: Yes.

**Comment by avencherus** [comment]
Isn't this more documenting the math, rather than the engine's function?

**Comment by aaronfranke** [comment]
It explains the purpose of the function and its possible outputs. I think this kind of documentation is great.

**Comment by asheraryam** [comment]
I think the explanation should mention that when the dot result is 1 then the two vectors are aligned and -1 for when they're in opposite directions. Explanations for dot product for game development usually start with 1, 0, -1 to make it more intuitive and and then talk about the values in between.

See this example: https://www.youtube.com/watch?v=2PrSUK1VrKA

Just saying that 0 -> 90 degrees and more than that is positive doesn't make it explicit enough what the other cases are and how the user should be thinking about making use of it.

**Comment by aaronfranke** [comment]
@asheraryam The range of -1 to 1 only applies for unit vectors, and therefore it's not the most general behavior of the method. I think it makes sense to first mention values greater/lesser than zero, because non-unit vectors will have a dot product different from 1 when aligned. For example, the dot product of `(1, 1)` and `(1, 1)` is `2`.

**Comment by asheraryam** [comment]
@aaronfranke Sure, but it could still have that extra information at the end, especially as using unit vectors with the dot product is the more common scenario.

So it would become something like

> When using unit vectors, the result will always be between [code]-1[/code] (180 degree angle) when the vectors are facing opposite directions, and [code]1[/code] (0 degree angle) when the vectors are aligned.

**Comment by akien-mga** [comment]
Needs a rebase on current `master` to fix CI.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Seems like the cherry-pick is not necessary, this was already added to the `3.2` branch somehow.

**Comment by aaronfranke** [comment]
@akien-mga [That was me](https://github.com/godotengine/godot/pull/40581/files#diff-48df85093746e1f516d425b6fb1ba16fR148), I just copied the master files on top of the 3.2 files (then ran doctool and fixed issues).

**Pull request merged**